File: tony-java-monolithic-demo/tony-api/src/main/java/com/tony/mono/api/controller/IndexController.java
package com.tony.mono.api.controller;

import com.tony.MonoResult;
import com.tony.PageQuery;
import com.tony.PageResult;
import com.tony.annotation.web.auth.NoLoginCheck;
import com.tony.jwt.JwtToken;
import com.tony.mono.db.po.User;
import com.tony.mono.db.service.UserService;
import com.tony.mono.dto.req.UserLoginReq;
import com.tony.web.utils.Servlets;
import io.swagger.v3.oas.annotations.Operation;
import kotlin.Pair;
import lombok.RequiredArgsConstructor;
import org.springframework.http.MediaType;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

import java.net.URL;
import java.util.Map;

/**
 * IndexController
 *
 * @author tangli
 * @date 2024/01/31 15:27
 */
@RequiredArgsConstructor
@Validated
@RestController
public class IndexController {

    private final UserService userService;

    @Operation(summary = "首页")
    @NoLoginCheck
    @GetMapping("/")
    public String index() {
        return "index";
    }

    @Operation(summary = "登录")
    @NoLoginCheck
    @PostMapping("/login")
    public MonoResult<String> login(
        @Validated
        @RequestBody final UserLoginReq req) {
        return MonoResult.ofMonoResult(JwtToken.gen(new Pair<>("userId", userService.login(req))));
    }

    @Operation(summary = "用户信息")
    @NoLoginCheck
    @PostMapping("/info")
    public void info() {
        // Yeah
    }

    @Operation(summary = "用户列表")
    @NoLoginCheck
    @PostMapping("/user/list")
    public PageResult<User> list(
        @Validated
        @RequestBody final PageQuery<String> req
    ) {
        return userService.list(req);
    }

    @Operation(summary = "requestHeaders")
    @NoLoginCheck
    @PostMapping("/request-headers")
    public Map<String,?> requestHeaders(){
        return Servlets.requestHeaders();
    }

    @Operation(summary = "responseHeaders")
    @NoLoginCheck
    @PostMapping("/response-headers")
    public Map<String,?> responseHeaders(){
        return Servlets.responseHeaders();
    }

    @Operation(summary = "origin")
    @NoLoginCheck
    @PostMapping("/origin")
    public MonoResult<String> origin(){
        return MonoResult.ofMonoResult(Servlets.origin());
    }

    @Operation(summary = "remoteIp")
    @NoLoginCheck
    @PostMapping("/remote-ip")
    public MonoResult<String> remoteIp(){
        return MonoResult.ofMonoResult(Servlets.remoteIp());
    }

    @Operation(summary = "url")
    @NoLoginCheck
    @PostMapping("/url")
    public URL url(){
        return Servlets.url();
    }

    @Operation(summary = "requestParsedMedia")
    @NoLoginCheck
    @PostMapping("/request-parsed-media")
    public MediaType requestParsedMedia(){
        return Servlets.requestParsedMedia();
    }

    @Operation(summary = "responseParsedMedia")
    @NoLoginCheck
    @PostMapping("/response-parsed-media")
    public MediaType responseParsedMedia(){
        return Servlets.responseParsedMedia();
    }
}

File: tony-monolithic-demo/tony-api/src/main/kotlin/com/tony/api/controller/IndexController.kt
package com.tony.api.controller

import com.tony.annotation.web.auth.NoLoginCheck
import com.tony.api.permission.NoPermissionCheck
import com.tony.db.service.UserService
import com.tony.dto.req.UserLoginReq
import com.tony.dto.resp.UserLoginResp
import com.tony.jwt.JwtToken
import com.tony.jwt.config.JwtProperties
import com.tony.utils.defaultZoneOffset
import com.tony.utils.toString
import com.tony.web.WebApp
import io.swagger.v3.oas.annotations.Operation
import java.time.LocalDateTime
import java.util.Locale
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
@Validated
class IndexController(
    private val userService: UserService,
    private val jwtProperties: JwtProperties,
) {
    @Operation(summary = "首页")
    @GetMapping("/")
    @NoLoginCheck
    @NoPermissionCheck
    fun index(): String =
        WebApp.appId

    @Operation(summary = "区域")
    @GetMapping("/locale")
    @NoLoginCheck
    @NoPermissionCheck
    fun locale(): String =
        Locale.getDefault().toLanguageTag()

    @Operation(summary = "zoneOffset")
    @GetMapping("/zone-offset")
    @NoLoginCheck
    @NoPermissionCheck
    fun zoneId(): String =
        defaultZoneOffset.toString()

    @Operation(summary = "时间戳")
    @GetMapping("/now")
    @NoLoginCheck
    @NoPermissionCheck
    fun now(): String =
        LocalDateTime.now().toString("yyyy-MM-dd HH:mm:ss")

    @Operation(summary = "登录")
    @NoLoginCheck
    @NoPermissionCheck
    @PostMapping("/login")
    fun login(
        @Validated
        @RequestBody
        loginReq: UserLoginReq,
    ): UserLoginResp {
        val token = JwtToken.gen("userId" to userService.login(loginReq).userId.toString())
        return UserLoginResp(token, token, LocalDateTime.now().minusMinutes(jwtProperties.expiredMinutes))
    }

    @Operation(summary = "空")
    @NoLoginCheck
    @NoPermissionCheck
    @GetMapping("/void")
    fun void() {
    }

    @Operation(summary = "异常")
    @NoLoginCheck
    @NoPermissionCheck
    @GetMapping("/exception")
    fun exception(): Unit =
        throw Exception("exception")
}
